packages/message-relayer/src/service.ts

```diff
@@ -549,4 +565,25 @@ export class MessageRelayerService extends BaseService<MessageRelayerOptions> {
     }
     this.logger.info('Message successfully relayed to Layer 1!')
   }
+
+  private async _getBlacklist(): Promise<void> {
+    try {
+      if (this.options.blacklistEndpoint) {
+        if (this.state.lastBlacklistPollingTimestamp === 0 || 
+          new Date().getTime() > this.state.lastBlacklistPollingTimestamp + this.options.blacklistPollingInterval
+        ) {
+          const response = await fetch(this.options.blacklistEndpoint);
+          const blacklist = await response.json();
+          this.state.lastBlacklistPollingTimestamp = new Date().getTime();
+          this.state.blacklist = blacklist;
+          this.logger.info('Found the blacklist', { blacklist })
+        }
+      } else {
+        this.state.blacklist = [];
+      }
+    } catch {
+      this.logger.info('Failed to fetch the blacklist')
+      this.state.blacklist = [];
+    }
```

Comment:
maybe this logic isn't good. If I understand it correct if you can't fetch the blacklist you're essentially allowing every message to go through. I think this might be dangerous. 
Instead, you should set it to `undefined`, so that message passing actually breaks and doesn't let messages through...

Reply:
No. The message relayer checks whether if the message in 7-days proof window. If it's not in the 7-days proof window, then it goes into `for (const message of messages) loop`. All messages in the `for loop` are intended to be relayed to L1. What I did is adding an another target address check in this `for loop`. If the target address in blacklist, then we don't relay the message.

Reply:
yes so if I understand correctly  - it's going to get relayed if `_getBlacklist` falls into the try *catch* clause.
But what if:
the message is NOT in the 7-days proof window, but it's blacklisted, but the blacklist endpoint is not reachable?


also, I don't know about typescript but in other languages, catching exceptions should be *precise* and *exact* to answer the question: what error are you trying to resolve (+ logging of the error that's missing).

Reply:
There are two types of messages in the for loop. The first one wants to be relayed to contracts that use the standard `OVM_L1CrossDomainMessenger`, which has the 7 days-proof window. The second one wants to be relayed to contracts that use the fast `OVM_L1FastCrossDomainMessenger `, which has 0 days-proof window. The message relayer is going to relay both of these messages, but the message using the fast `OVM_L1FastCrossDomainMessenger` can't be successfully relayed.  The L1 contract only receives the messages from the right `OVM_L1CrossDomainMessenger `. The message relayer uses the `OVM_L1CrossDomainMessenger `, so it can't successfully relay messages to contracts that use the `OVM_L1FastCrossDomainMessenger `. 

There are two reasons that I used the catching exceptions:
1. I don't want the message relayer to stop relaying the messages because of the backend API error. If we set it to `undefined`, the message relayer basically won't relay any messages.
2. We can get the notification when it meets error.